synapse/federation/transport/client.py

```diff
@@ -102,6 +102,10 @@ async def get_room_state(
             destination,
             path=path,
             args={"event_id": event_id},
+            # This can take a looooooong time for large rooms. Give this a generous
+            # timeout, to avoid the partial state resync timing out early and trying
+            # a bunch of servers who haven't see our join yet.
+            timeout=600,
```

Comment:
Thanks, it is. Which is confusing, because `default_timeout` isi in seconds:

https://github.com/matrix-org/synapse/blob/d8cc86eff484b6f570f55a5badb337080c6e4dcd/synapse/http/matrixfederationclient.py#L369